src/org/zaproxy/zap/extension/fieldenumeration/FieldEnumeration.java

```diff
@@ -0,0 +1,661 @@
+
+		try {
+	        	encodedValue = (escaped) ? value : URLEncoder.encode(value, StandardCharsets.UTF_8.name());
+		} catch (UnsupportedEncodingException ignore) {
+		}
+
+		System.out.println("Encoder " + encodedValue);
+        	NameValuePair pair;
+        	boolean isAppended = true;
+
+        	for (int i = 0; i < getParamList().size(); i++) {
+            		pair = getParamList().get(i);
+			
+			if (pair.getName() == name) {
+				isAppended = paramAppend(sb, getEscapedName(msg, name), encodedValue, parser);
+			}
+			else {
+				try {
+       	   				isAppended = paramAppend(sb, getEscapedName(msg, pair.getName()), URLEncoder.encode(pair.getValue(), StandardCharsets.UTF_8.name()), parser);
+				} catch (UnsupportedEncodingException ignore) {
+				}
+			}
+
+            		if (isAppended && i < getParamList().size() - 1) {
+                		sb.append(parser.getDefaultKeyValuePairSeparator());
+            		}
+        	}
+
+        	if (sb.length() == 0) {
+             //No original query string
+            		sb.append(encodedValue);
+        	}
+        	String query = sb.toString();
+		System.out.println("Query " + query);
+		msg.getRequestBody().setBody(query);
+        	return query;
+
+	}
+
+	private boolean paramAppend(StringBuilder sb, String name, String value, ParameterParser parser) {
+       		boolean isEdited = false;
+
+        	if (name != null) {
+            		sb.append(name);
+            		isEdited = true;
+        	}
+
+        	if (value != null) {
+            		sb.append(parser.getDefaultKeyValueSeparator());
+            		sb.append(value);
+            		isEdited = true;
+        	}
+
+        	return isEdited;
+        } 
+
+	private void initialize() {
+        	this.setTitle(Constant.messages.getString("fieldenumeration.field.popup"));
+		this.setContentPane(getJPanel());
+		
+		if (Model.getSingleton().getOptionsParam().getViewParam().getWmUiHandlingOption() == 0) {
+                	this.setSize(400, 400);
+	        }
+        	this.addWindowListener(new java.awt.event.WindowAdapter() {
+                	@Override
+                	public void windowOpened(java.awt.event.WindowEvent e) {
+                	}
+
+                	@Override
+                	public void windowClosing(java.awt.event.WindowEvent e) {
+                	}
+        	});
+
+                pack();
+        }
+
+
+	public class ButtonListenerOk implements ActionListener
+        {
+            @Override
+            public void actionPerformed(ActionEvent arg0)
+            {
+                StringBuilder stringB = new StringBuilder();
+		stringB.append("Regular Exp: ").append(regex.getText());
+                for(JTextField k : listOfTextFields){
+                        stringB.append(k.getText()).append(" \n");
+                }
+                JOptionPane.showMessageDialog(null, stringB.toString());
+            }
+        }
+	 
+	 private JFrame getResult() {
+		result.setLayout(new GridBagLayout());
```

Comment:
It's probably better to use another `result` frame, to avoid changing existing enumeration results (if not, it should be taken care to not add multiple labels/tables/components).

Reply:
This was the issue, I faced.